### PR TITLE
[NO-ISSUE] refactor how we login to jolokia

### PR DESCRIPTION
**Pull request by lavocatt**
To minimize code duplication, extract the jolokia login to a new component JolokiaAuthentification. This component does the login and creates a context for its children components.

Now a developer can use this pattern to ensure the component someComponent will get in a logged in context.

```typescript
const AuthenticatedComponent: FC = () => {
  const brokerName = 'someName';
  const namespace = 'someNamespace';
  const podOrdinal = 0;
  const { brokerCr } = UseGetBrokerCR(brokerName, namespace);
  return (
    <JolokiaAuthentification brokerCR={brokerCr} podOrdinal={podOrdinal}>
      <SomeComponent />
    </JolokiaAuthentification>
  );
}
```

### CODE REVIEW

File: src/jolokia/components/JolokiaAuthentification.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import { FC, useState } from 'react';
+import { OpenAPI as OpenAPIConfig } from '../../openapi/jolokia/requests/core/OpenAPI';
+import {
+  useGetApiServerBaseUrl,
+  useJolokiaLogin,
+} from '../../jolokia/customHooks';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { AuthContext } from '../context';
+import { BrokerCR } from '../..//k8s/types';
```
Comment (Msarawan) [inline src/jolokia/components/JolokiaAuthentification.tsx:9]: Hello @lavocatt , just a small minor update could you please remove the extra `/`  in the import of BrokerCR.

File: src/addresses/address-details/AddressDetails.container.tsx [hunk 2]
```diff
@@ -42,93 +26,16 @@ export const AddressDetailsPage: FC = () => {
...
-          <Button key="cancel" variant="link" onClick={handleModalToggle}>
-            {t('cancel')}
-          </Button>,
-        ]}
-      >
-        <TextContent>
-          <Text component={TextVariants.h6}>{t('login_failed_message')}</Text>
-        </TextContent>
-      </Modal>
+    <JolokiaAuthentification brokerCR={brokerDetails} podOrdinal={podOrdinal}>
```
Comment (gaohoward) [inline src/addresses/address-details/AddressDetails.container.tsx:38]: typo: JolokiaAuthentication ?

File: src/jolokia/context.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { createContext } from 'react';
+
+export type jolokiaLoginSource = 'api' | 'session';
+
+export type JolokiaLogin = {
+  isSucces: boolean;
```
Comment (Msarawan) [inline src/jolokia/context.ts:6]: Currently, the boolean variable is spelled as `isSucces` in several places. Could you please update the spelling to `isSuccess` for clarity and correctness? This change will help to maintain consistency and readability in the codebase.

File: src/jolokia/context.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  isLoading: boolean;
+  isError: boolean;
+  token: string;
+  source: jolokiaLoginSource;
+};
+
+export const AuthContext = createContext<JolokiaLogin>({
+  token: '',
+  isLoading: true,
+  isSucces: false,
```
Comment (Msarawan) [inline src/jolokia/context.ts:16]: same here..

File: src/addresses/address-details/AddressDetails.container.tsx (hunk 2 above)
Comment (lavocatt) [reply src/addresses/address-details/AddressDetails.container.tsx:38]: Ah right! Authentification is French :sweat_smile:

**Review by Msarawan — APPROVED** [review]
LGTM. Good work Thomas.

**Pull request merged**